sense/stick/stick.go

```diff
@@ -1,73 +1,115 @@
 package stick
 
 import (
-	"errors"
-	"fmt"
-	"io/ioutil"
 	"os"
-	"path/filepath"
 	"strings"
+	"syscall"
+	"unsafe"
 )
 
-// Joystick events
-var (
-	Right = 0
-	Left  = 1
-	Up    = 2
-	Down  = 3
-	Enter = 4
+const (
+	Enter = 28
+	Up    = 103
+	Left  = 105
+	Right = 106
+	Down  = 108
 )
 
-// the joy stick
-var joyStickDevice string
+type Device struct {
+	f      *os.File
+	Events chan Event
+}
 
-func init() {
-	var err error
-	joyStickDevice, err = getDevice("Raspberry Pi Sense HAT Joystick")
+func Open(name string) (*Device, error) {
+	f, err := os.Open(name)
 	if err != nil {
-		panic(err)
+		return nil, err
 	}
+
+	d := &Device{f, make(chan Event, 4)}
```

Comment:
There is a slim chance that more than one Event is read in a single call to `d.pollEvents`.
The eventBufferSize in [github.com/jteeuwen/evdev](https://github.com/jteeuwen/evdev) is set to 64, but i opted for a much lower number since it is unlikely that you could physically input that many events (at close to the same time) using the Sense HAT joystick.